engines/msaccess.py

```diff
@@ -56,12 +58,10 @@ def insert_data_from_file(self, filename):
         """Perform a bulk insert."""
         self.get_cursor()
         ct = len([True for c in self.table.columns if c[1][0][:3] == "ct-"]) != 0
-        if ((self.table.cleanup.function == no_cleanup and not self.table.fixed_width and
-             self.table.header_rows < 2)
-            and (self.table.delimiter in ["\t", ","])
-            and not ct
-            and (not hasattr(self.table, "do_not_bulk_insert") or not self.table.do_not_bulk_insert)
-            ):
+        if (not (not (self.table.cleanup.function == no_cleanup and not self.table.fixed_width and
+                      self.table.header_rows < 2) or not (self.table.delimiter in ["\t", ","]) or ct) and
+                (not hasattr(self.table, "do_not_bulk_insert") or not self.table.do_not_bulk_insert)):
```

Comment:
This conditional is really confusing to read since it requires tracking double and even triple negatives to figure out what's happening. Even if this is technically PEP 8 I think we need to find a better way of writing the logic. The original form is better, but I'm happy to look at an alternative.